client/apps/edit/components/section_image_collection/index.coffee

```diff
@@ -3,151 +3,55 @@
 #
 
 _ = require 'underscore'
-gemup = require 'gemup'
 React = require 'react'
-sd = require('sharify').data
-icons = -> require('./icons.jade') arguments...
-Autocomplete = require '../../../../components/autocomplete/index.coffee'
-Artwork = require '../../../../models/artwork.coffee'
 imagesLoaded = require 'imagesloaded'
-RichTextCaption = React.createFactory require '../../../../components/rich_text_caption/index.coffee'
-UrlArtworkInput = React.createFactory require '../section_image_set/url_artwork_input.coffee'
-
-{ div, section, h1, h2, span, img, header, input, a, button, p, ul, li, strong, nav } = React.DOM
-{ resize } = require '../../../../components/resizer/index.coffee'
+DisplayArtwork = React.createFactory require './components/artwork.coffee'
+DisplayImage = React.createFactory require './components/image.coffee'
+Controls = React.createFactory require './components/controls.coffee'
+{ div, section, ul, li } = React.DOM
 
 module.exports = React.createClass
   displayName: 'SectionImageCollection'
 
   getInitialState: ->
-    images: @props.section.get('images') or []
-    layout: @props.section.get('layout') or 'overflow_fillwidth'
     progress: null
 
   componentDidMount: ->
     @$list = $(@refs.images)
-    @setupAutocomplete()
     imagesLoaded @$list, =>
-      if @state.images.length > 1 and @state.layout is 'overflow_fillwidth'
-        @toggleFillwidth()
-      else
-        @$list.animate({opacity: 1}, 500)
-
-  componentWillUnmount: ->
-    @autocomplete.remove()
-
-  componentWillUpdate: ->
-    @props.section.set images: @state.images if @state.images.length > 0
-
-  setupAutocomplete: ->
-    $el = $(@refs.autocomplete)
-    @autocomplete = new Autocomplete
-      url: "#{sd.ARTSY_URL}/api/search?q=%QUERY"
-      el: $el
-      filter: (res) ->
-        vals = []
-        for r in res._embedded.results
-          if r.type?.toLowerCase() == 'artwork'
-            id = r._links.self.href.substr(r._links.self.href.lastIndexOf('/') + 1)
-            vals.push
-              id: id
-              value: r.title
-              thumbnail: r._links.thumbnail?.href
-        return vals
-      templates:
-        suggestion: (data) ->
-          """
-            <div class='esis-suggestion' \
-                 style='background-image: url(#{data.thumbnail})'>
-            </div>
-            #{data.value}
-          """
-      selected: @onSelect
-    _.defer -> $el.focus()
-
-  onSelect: (e, selected) ->
-    new Artwork(id: selected.id).fetch
-      success: (artwork) =>
-        newImages = @state.images.concat [artwork.denormalized()]
-        @setState images: newImages
-        @props.section.set images: newImages
-        $(@refs.autocomplete).val('').focus()
-        @toggleFillwidth() if @state.images.length > 1
+      @onChange()
+      @$list.animate({opacity: 1}, 250) if @props.section.get('layout') != 'column_width'
```

Comment:
If possible it's often nicer to avoid using jQuery to manipulate the DOM with React—even in the case of animations. A common way to do this is to use CSS transitions that you toggle in `render`. e.g.

````coffeescript
render: ->
  # ...
  ul {
    className: 'esic-images-list',
    'data-layout': @props.section.get('layout'),
    ref: 'images'
  },
    # ...
````

````stylus
.esic-images-list
  transition opacity 2.5ms
  &:not([data-layout="column_width"])
    opacity 1
````

Not a blocker for this PR and there are legit reasons to do it this way too. Just food for thought.